base/array.jl

```diff
@@ -669,30 +669,34 @@ _array_for(::Type{T}, itr, isz) where {T} = _array_for(T, isz, _similar_shape(it
     collect(collection)
 
 Return an `Array` of all items in a collection or iterator. For dictionaries, returns
-`Vector{Pair{KeyType, ValType}}`. If the argument is array-like or is an iterator with the
-[`HasShape`](@ref IteratorSize) trait, the result will have the same shape
+a `Vector` of `key=>value` [pairs](@ref Pair). If the argument is array-like or is an iterator
+with the [`HasShape`](@ref IteratorSize) trait, the result will have the same shape
 and number of dimensions as the argument.
 
-Used by comprehensions to turn a generator into an `Array`.
+Used by [comprehensions](@ref man-comprehensions) to turn a [generator expression](@ref man-generators)
+into an `Array`. Thus, *on generators*, the bracket notation can be used instead of calling `collect`,
```

Comment:
```suggestion
into an `Array`. Thus, *on generators*, the square-brackets notation may be used instead of calling `collect`,
```
In British (and Indian) English, brackets often refer to parentheses, so it's better to be clearer about this.